crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2381,9 +2445,17 @@ pub fn transaction_build(
         } else {
             // if transaction is already mined, gas price is considered base fee + priority fee: the
             // effective gas price.
-            let base_fee = base_fee.unwrap_or(0);
+            let base_fee = base_fee.unwrap_or(0u128);
             let max_priority_fee_per_gas = transaction.max_priority_fee_per_gas.unwrap_or(0);
             transaction.gas_price = Some(base_fee.saturating_add(max_priority_fee_per_gas));
+
+            // if the tx is eip4844 we need to also add the max fee per blob gas.
+            if eth_transaction.is_eip4844() {
+                transaction.gas_price = transaction
+                    .gas_price
+                    .map(|g| g.checked_add(transaction.max_fee_per_blob_gas.unwrap_or_default()))
+                    .unwrap_or(Some(u128::MAX));
+            }
```

Comment:
Not sure if this is correct, blob fee is charged from blob gas and `gas_price` corresponds to the execution gas?

Reply:
Yes this was a mistake